proposals/NNNN-shader-execution-reordering.md

```diff
@@ -0,0 +1,1388 @@
+ HitObject_WorldRayOrigin    | `float` | vector | Returns the ray origin in world space.
+ HitObject_WorldRayDirection | `float` | vector | Returns the ray direction in world space.
+ HitObject_ObjectRayOrigin   | `float` | vector | Returns the ray origin in object space.
+ HitObject_ObjectRayDirection| `float` | vector | Returns the ray direction in object space.
+ HitObject_ObjectToWorld3x4  | `float` | matrix | Returns the object to world space transformation matrix in 3x4 form.
+ HitObject_ObjectToWorld4x3  | `float` | matrix | Returns the object to world space transformation matrix in 4x3 form.
+ HitObject_WorldToObject3x4  | `float` | matrix | Returns the world to object space transformation matrix in 3x4 form.
+ HitObject_WorldToObject4x3  | `float` | matrix | Returns the world to object space transformation matrix in 4x3 form.
+ HitObject_GeometryIndex     | `i32`   | scalar | Returns the geometry index committed on hit.
+ HitObject_InstanceIndex     | `i32`   | scalar | Returns the instance index committed on hit.
+ HitObject_InstanceID        | `i32`   | scalar | Returns the instance id committed on hit.
+ HitObject_PrimitiveIndex    | `i32`   | scalar | Returns the primitive index committed on hit.
+ HitObject_HitKind           | `i32`   | scalar | Returns the HitKind of the hit.
+ HitObject_ShaderTableIndex  | `i32`   | scalar | Returns the shader table index set for this HitObject.
+
+
+```DXIL
+declare T @dx.op.hitObject_StateScalar.T(
+    i32,                      ; opcode
+    %dx.types.HitObject)      ; hit object
+    nounwind readnone
+```
+The overload `T` is a valid return type as listed above.
+
+```DXIL
+declare float @dx.op.hitObject_StateVector.f32(
+    i32,                      ; opcode
+    %dx.types.HitObject,      ; hit object
+    i32)                      ; index
+    nounwind readnone
+```
+```DXIL
+declare float @dx.op.hitObject_StateMatrix.f32(
+    i32,                      ; opcode
+    %dx.types.HitObject,      ; hit object
+    i32,                      ; row
+    i32)                      ; column
+    nounwind readnone
+```
+
+#### HitObject_LoadLocalRootTableConstant
+
+Returns the root table constant for this HitObject and offset.
+
+```DXIL
+declare i32 @dx.op.hitObject_LoadLocalRootTableConstant(
+    i32,                           ; opcode
+    %dx.types.HitObject,           ; hit object
+    i32)                           ; offset
+    nounwind readonly
+```
+
+#### HitObject_Attributes
+
+Copies the attributes set for this HitObject to the provided buffer.
+
+```DXIL
+declare void @dx.op.hitObject_Attributes.AttrT(
+    i32,                           ; opcode
+    %dx.types.HitObject,           ; hit object
+    AttrT*)                        ; attributes
+    nounwind argmemonly
+```
+
+`AttrT` is the user-defined intersection attribute struct type. See `ReportHit` for definition.
+
+#### HitObject_SetShaderTableIndex
```

Comment:
This is out of order with the table above

Reply:
Moved.